presto-bigquery/src/main/java/com/facebook/presto/plugin/bigquery/BigQueryEmptySplitPageSource.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.plugin.bigquery;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.common.Page;
+import com.facebook.presto.common.PageBuilder;
+import com.facebook.presto.spi.ConnectorPageSource;
+import com.google.common.collect.ImmutableList;
+
+import java.io.IOException;
+
+public class BigQueryEmptySplitPageSource
```

Comment:
let's merge all logic in BigQueryEmptySplitPageSource
into BigQueryResultPageSource
we do not need a class for empty page source
BigQueryResultPageSource should cover everything

Reply:
can we do this logic change in a separate PR? I do have a number of follow-up items as future roadmap for bigquery connector and this can be an addition to that list

Reply:
I am inclined to do the merge in this commit. None of other connectors have separate class for empty page source.

Reply:
@highker @beinan @qqibrow what do you think of keeping BigQueryEmptySplitPageSource as a separate class? I am inclined to merge BigQueryEmptySplitPageSource into BigQueryResultPageSource to simplify logic, but seems we already have HiveEmptySplitPageSource. Do you think it is fine to keep it as a first step?

Reply:
As the first commit of this commit, I'm ok to keep the BigQueryEmptySplitPageSource.  As @zhenxiao  said, we could use BigQueryResultPageSource to cover everything in the future.